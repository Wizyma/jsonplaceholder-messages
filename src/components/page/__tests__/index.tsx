import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Component from '..';

beforeEach(() => {
  cleanup();
});

describe(`${Component.name} should pass all test`, () => {
  it('renders children correctly', () => {
    const { asFragment, getByText } = render(<Component children={<p>Hello World</p>} />)
    const child = getByText(/Hello World/i);
    expect(child).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot()
  })
})