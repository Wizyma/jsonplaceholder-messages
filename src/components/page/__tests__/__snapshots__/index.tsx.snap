// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should pass all test renders children correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <main
      class="MuiContainer-root makeStyles-main-2 MuiContainer-maxWidthSm"
    >
      <div
        style="margin-bottom: 50px;"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter"
        >
          Messenger
        </h2>
      </div>
      <p>
        Hello World
      </p>
    </main>
    <footer
      class="makeStyles-footer-3"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter"
        >
          Andr√© Gomes - jsonplaceholder-messages
        </h6>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
